Simprints/Android-Simprints-ID
# MS-668 Code style improvements

## PR structure 
1. Commit with the KtLint and general formatting configuration 
2. Commit with fixes for issues that cannot be automatically fixed by the KtLint. The report can be generated by running 
`ktlint '**/*.kt' '!**/generated/**.kt' --format --reporter=html,output=build/ktlint-report.html`. The only unresolved issues are long lines. Fixes mostly fall into one of these categories:
    * Wildcard imports
    * Incorrect property name case
    * File contains a single class with a different name
    * Consecutive comments of different kinds
    * Empty file

3. Optional commit with automatically generated fixes (may not be present all of the time). I might not include this commit in some iterations to avoid polluting the UI, the steps below are the recommended way to do the review.

## How to review

Do not review this code in Github as the UI is not capable of handling 20k changes, instead do the following: 

1. Install KtLint CLI - https://pinterest.github.io/ktlint/latest/install/cli/ (do not install the plugin right now)
2. Check out the branch, but do a hard reset to the second commit
3. Run KtLint CLI command above
4. Review the changes in your favourite GIT diffing tool
5. If you find any objectable formatting - check if there is a rule that can disable this formatting - https://pinterest.github.io/ktlint/latest/rules/standard/ and propose a configuration update as a comment in the `.editorconfig` file (to allow discussion thread for specific proposal).
